common/smoke-test/Update-SmokeTests.ps1

```diff
@@ -0,0 +1,155 @@
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $TestApplicationId,
+
+  [Parameter()]
+  [string] $TestApplicationSecret,
+
+  [Parameter()]
+  [string] $TestApplicationOid,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string] $TenantId,
+
+  [Parameter(ParameterSetName = 'Provisioner')]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $SubscriptionId,
+
+  [Parameter()]
+  [string] $Location = '',
+
+  [Parameter()]
+  [string] $Environment = 'AzureCloud',
+
+  [Parameter()]
+  [hashtable] $AdditionalParameters,
+
+  [Parameter()]
+  [switch] $CI = ($null -ne $env:SYSTEM_TEAMPROJECTID),
+
+  # Captures any arguments not declared here (prevents no parameter errors)
+  [Parameter(ValueFromRemainingArguments = $true)]
+  $RemainingArguments
+)
+function SetEnvironmentVariable {
+  param(
+    [string] $Name,
+    [string] $Value
+  )
+
+  if ($CI) {
+    Write-Host "##vso[task.setvariable variable=_$Name;issecret=true;]$($Value)"
+    Write-Host "##vso[task.setvariable variable=$Name;]$($Value)"
+  }
+  else {
+    Write-Verbose "Setting local environment variable: $Name = ***"
+    Set-Item -Path "env:$Name" -Value $Value
+  }
+}
+
+pushd ../../
+
+Write-Verbose "Reading manifest..."
+$manifest = (Get-Content -Path common/smoke-test/samples-manifest.json | ConvertFrom-Json)
+
+$deployedServiceDirectories = @{ }
+$runManifest = @()
+$dependencies = New-Object 'system.collections.generic.dictionary[string,string]'
+$baseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+
+# Use the same resource group name that New-TestResources.ps1 generates
+SetEnvironmentVariable -Name 'AZURE_RESOURCEGROUP_NAME' -Value "rg-$baseName"
+
+foreach ($entry in $manifest) {
+  try {
+    Write-Verbose "Deploying resources for $($entry.name)..."
+    if ($deployedServiceDirectories.ContainsKey($entry.resourcesDirectory) -ne $true) {
+
+      # Force -CI to false here. This is to have the script make use of
+      # -BaseName so that all resources are created within the same resource
+      # group for easier cleanup. All created environment variables are returned
+      # to $deployOutput so they can be set in the user's context or in CI.
+      $deployOutput = eng/common/TestResources/New-TestResources.ps1 `
+        -BaseName  $baseName `
+        -ServiceDirectory $entry.resourcesDirectory `
+        -TestApplicationId $TestApplicationId `
+        -TestApplicationSecret $TestApplicationSecret `
+        -ProvisionerApplicationId $TestApplicationId `
+        -ProvisionerApplicationSecret $TestApplicationSecret `
+        -TestApplicationOid $TestApplicationOid `
+        -TenantId $TenantId `
+        -SubscriptionId $SubscriptionId `
+        -Location $Location `
+        -Environment $Environment `
+        -AdditionalParameters $AdditionalParameters `
+        -Force `
+        -Verbose `
+        -CI:$false
```

Comment:
Lets see if we can just pass through CI and if we need to change something specific then lets add an additional parameter for that.